internal/guest/runtime/hcsv2/spec.go

```diff
@@ -202,25 +191,13 @@ func getGroup(spec *oci.Spec, filter func(user.Group) bool) (user.Group, error)
 func applyAnnotationsToSpec(ctx context.Context, spec *oci.Spec) error {
 	// Check if we need to override container's /dev/shm
 	if val, ok := spec.Annotations[annotations.LCOWDevShmSizeInKb]; ok {
-		sz, err := strconv.ParseInt(val, 10, 64)
+		mt, err := devShmMountWithSize(val)
 		if err != nil {
-			return errors.Wrap(err, "/dev/shm size must be a valid integer")
-		}
-		if sz <= 0 {
-			return errors.Errorf("/dev/shm size must be a positive integer, got: %d", sz)
-		}
-
-		// Use the same options as in upstream https://github.com/containerd/containerd/blob/0def98e462706286e6eaeff4a90be22fda75e761/oci/mounts.go#L49
-		size := fmt.Sprintf("size=%dk", sz)
-		mt := oci.Mount{
-			Destination: "/dev/shm",
-			Type:        "tmpfs",
-			Source:      "shm",
-			Options:     []string{"nosuid", "noexec", "nodev", "mode=1777", size},
+			return err
 		}
 		spec.Mounts = removeMount("/dev/shm", spec.Mounts)
-		spec.Mounts = append(spec.Mounts, mt)
-		log.G(ctx).WithField("size", size).Debug("set custom /dev/shm size")
+		spec.Mounts = append(spec.Mounts, *mt)
+		log.G(ctx).WithField("size", fmt.Sprintf("size=%sk", val)).Debug("set custom /dev/shm size")
```

Comment:
Logging the value as `size=Nk` is kind of unhelpful. Can we just put the actual value in there, and put the units in the field name?

Reply:
let me know if the updated version works.

Reply:
We try to avoid spaces in field names, just `sizeKB` seems fine though.